integrations/sharepoint/src/SharepointClient.ts

```diff
@@ -0,0 +1,441 @@
+          const extension = path.extname(fileName);
+          if (!SUPPORTED_FILE_EXTENSIONS.includes(extension)) {
+            this.log(`File extension not supported for file: ${fileName}`);
+            break;
+          }
+          const content = await this.downloadFile(fileName);
+          await this.botpressKB.addFile(change.ItemId.toString(), fileName, content);
+          break;
+        }
+        // Update
+        case 2: {
+          try {
+            this.log(`Updating item: ${change.ItemId}`);
+            const updatedFileName = await this.getFileName(change.ItemId);
+            if (!updatedFileName) {
+              this.log(`File does not exist for item: ${change.ItemId}`);
+              break;
+            }
+            this.log(`Updated file name: ${updatedFileName}`);
+            const updatedContent = await this.downloadFile(updatedFileName);
+            // Delete the existing file and add the updated file
+            await this.botpressKB.deleteFile(change.ItemId.toString());
+            await this.botpressKB.addFile(change.ItemId.toString(), updatedFileName, updatedContent);
+          } catch (e) {
+            this.log(`Error updating file: ${change.ItemId}: ${e}`);
+          }
+          break;
+        }
+        // Delete
+        case 3: {
+          try {
+            this.log(`Deleting item: ${change.ItemId}`);
+            await this.botpressKB.deleteFile(change.ItemId.toString());
+          } catch (e) {
+            this.log(`Error deleting file: ${change.ItemId}: ${e}`);
+          }
+          break;
+        }
+        // Rename
+        case 4: {
+          try {
+            this.log(`Renaming item: ${change.ItemId}`);
+            const renamedFileName = await this.getFileName(change.ItemId);
+            if (!renamedFileName) {
+              this.log(`File does not exist for item: ${change.ItemId}`);
+              break;
+            }
+            this.log(`Renamed file name: ${renamedFileName}`);
+            const content = await this.downloadFile(renamedFileName);
+            // Delete the existing file and add the updated file
+            await this.botpressKB.deleteFile(change.ItemId.toString());
+            await this.botpressKB.addFile(change.ItemId.toString(), renamedFileName, content);
+          } catch (e) {
+            this.log(`Error renaming file: ${change.ItemId}: ${e}`);
+          }
+          break;
+        }
+        default: {
+          this.log(`Change type not supported (yet): ${change.ChangeType}`);
+          break;
+        }
+      }
+      this.log(`Processed change type ${change.ChangeType} for item: ${change.ItemId}`);
+    }
+
+    return newChangeToken;
+  }
+}
+
+/**
+ * Factory function to create a SharepointClient
+ * @param integrationConfiguration - The integration configuration object
+ * @param botpressKB - The BotpressKB object
+ * @returns - The SharepointClient
+ */
+export const createSharepointClient = (
```

Comment:
**question:** Do I understand correctly that this function wraps the class constructor without any additional logic?

Reply:
yep, I had some logic in it earlier which was removed. Now this is redundant. removed it!